indico/modules/designer/controllers.py

```diff
@@ -0,0 +1,276 @@
+        'height': {'type': 'integer', 'minimum': 0},
+        'background_position': {'type': 'string'},
+        'items': {
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'id': {'type': 'integer'},
+                    'text': {'type': 'string'},
+                    'x': {'type': 'number', 'minimum': 0},
+                    'y': {'type': 'number', 'minimum': 0},
+                    'width': {'type': 'integer', 'minimum': 0},
+                    'color': {'type': 'string'},
+                    'font_family': {'type': 'string'},
+                    'font_size': {'type': 'string'},
+                    'text_align': {'type': 'string'},
+                    'bold': {'type': 'boolean'},
+                    'italic': {'type': 'boolean'}
+                }
+            },
+            'required': ['id', 'text', 'x', 'y', 'width', 'color', 'font_family', 'font_size', 'text_align',
+                         'bold', 'italic']
+        },
+    },
+    'required': ['width', 'height', 'background_position', 'items']
+}
+
+
+def _render_template_list(target, event):
+    tpl = get_template_module('designer/_list.html')
+    return tpl.render_template_list(target.designer_templates, target, event=event,
+                                    inherited_templates=get_inherited_templates(target))
+
+
+class TemplateDesignerMixin:
+    @property
+    def object_type(self):
+        """Figure out whether we're targetting an event or category, based on URL info."""
+        return request.view_args.get('object_type')
+
+    @property
+    def owner_dict(self):
+        return {'event_new': self.event_new} if self.object_type == 'event' else {'category': self.category}
+
+    @property
+    def event_or_none(self):
+        return self.owner if self.object_type == 'event' else None
+
+    def _render_template(self, tpl_name, **kwargs):
+        if self.object_type == 'event':
+            return WPEventManagementDesigner.render_template(tpl_name, self.owner.as_legacy, target=self.owner,
+                                                             event=self.owner, **kwargs)
+        else:
+            return WPCategoryManagementDesigner.render_template(tpl_name, self.owner, 'designer', target=self.owner,
+                                                                **kwargs)
+
+
+class OwnerFromURLMixin(TemplateDesignerMixin):
+    @property
+    def owner(self):
+        return self.event_new if self.object_type == 'event' else self.category
+
+
+class OwnerFromTemplateMixin(TemplateDesignerMixin):
+        @property
+        def owner(self):
+            return self.template.owner
+
+
+class TemplateListMixin(OwnerFromURLMixin):
+    def _process(self):
+        return self._render_template('list.html', inherited_templates=get_inherited_templates(self.owner))
+
+
+class SpecificTemplateMixin(OwnerFromTemplateMixin):
+
```

Comment:
PEP257 checks for some of that stuff. Not sure if that's included. Not in my config, though.